packages/connect-form/src/components/connect-form-actions.tsx

```diff
@@ -11,37 +11,76 @@ import {
   ConnectionFormError,
   ConnectionFormWarning,
 } from '../utils/validation';
+import type { ConnectionInfo } from 'mongodb-data-service';
 
 const formActionStyles = css({
   borderTop: `1px solid ${uiColors.gray.light2}`,
   paddingLeft: spacing[4],
   paddingRight: spacing[4],
 });
 
+const formActionButtonStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  '& div:first-child': {
+    marginRight: spacing[2],
+  },
+});
+
 const formButtonsStyles = css({
   paddingTop: spacing[3],
   paddingBottom: spacing[3],
   textAlign: 'right',
 });
 
 function ConnectFormActions({
+  initialConnectionInfo,
   errors,
   warnings,
   onConnectClicked,
+  onSaveClicked,
+  saveDisabled,
 }: {
-  onConnectClicked: () => void;
+  initialConnectionInfo: ConnectionInfo;
   errors: ConnectionFormError[];
   warnings: ConnectionFormWarning[];
+  onConnectClicked: () => void;
+  onSaveClicked: () => void;
+  saveDisabled: boolean;
 }): React.ReactElement {
   return (
     <div className={formActionStyles}>
       {warnings.length ? <WarningSummary warnings={warnings} /> : ''}
       {errors.length ? <ErrorSummary errors={errors} /> : ''}
-
-      <div className={formButtonsStyles}>
-        <Button variant={ButtonVariant.Primary} onClick={onConnectClicked}>
-          Connect
-        </Button>
+      <div className={formActionButtonStyles}>
+        {!initialConnectionInfo.favorite && (
+          <div className={formButtonsStyles}>
```

Comment:
this wrapper is not necessary anymore, we can move the `padding` to  `formActionButtonStyles` and drop this, so we don't bloat the layout.

Reply:
I think we need the wrapper because it's the flex container that will place the buttons horizontally (before them there is the warning/errors banner that should stack vertically instead)

Reply:
What I mean is that you need 1 wrapper but not 2. right now we have:

```
<div className={formActionButtonStyles}>
   <div className={formButtonsStyles}><button></div>
   <div className={formButtonsStyles}><button></div>
</div> 
```

we can do just:

```
<div className={formActionButtonStyles}>
   <button>
   <button>
</div> 
```

Reply:
got it, fixed!